climt/_components/surface_ice.py

```diff
@@ -317,7 +332,7 @@ def calculate_new_ice_temperature(self, rho, specific_heat, kappa,
 
         # Set flux condition if temperature is below melting point,
         # and dirichlet condition above melting point
-        if surf_temp < self._temp_melt:
+        if surf_temp < self._temp_melt - self._epsilon:
```

Comment:
Avoid "temp" as an abbreviation for temperature - it's easily confused with "temporary". Instead use "temperature" if only used outside of equations or "T" if used in long expressions where variable name length matters.